mypy/typeanal.py

```diff
@@ -813,7 +813,10 @@ def analyze_callable_type(self, t: UnboundType) -> Type:
                     return AnyType(TypeOfAny.from_error)
                 ret = maybe_ret
         else:
-            self.fail('Please use "Callable[[<parameters>], <return type>]" or "Callable"', t)
+            if self.options.disallow_any_generics:
+                self.fail('Please use "Callable[[<parameters>], <return type>]" or "Callable[..., Any]"', t)
```

Comment:
```suggestion
                self.fail('Please use "Callable[[<parameters>], <return type>]"', t)
```
Let's remove the suboptimal suggestion here

Reply:
My 2 cents: I actually want to keep the error message with both suggestions, because the 'suboptimal' suggestion is _exactly_ what I wanted to know. So for me it was very helpful (albeit unfortunately misleading, which is why I've corrected it). Just to emphasize that some users _will_ get value from the not-concise message.

That said, if you still want me to remove that part I'll gladly do so.

Reply:
@JelleZijlstra just to double-check: I lined out above why I wanted to keep that message as is. Do you still want me to change it?

Reply:
I'd still lean towards removing it. Error messages should be short and provide the most useful option, and `Callable[..., Any]` generally isn't what we should steer people towards.

Reply:
Okay, I've incorporated your suggestion